### PR TITLE
RD-6920 Add environment blueprint generation endpoint

### PR DESCRIPTION
**Pull request by kubama**
## Description
For now I just copied existing solutions from terraform endpoint. The next task is to extract common parts.

## Screenshots / Videos
N/A

## Checklist

- [X] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [X] I added proper labels to this PR.

## Tests
Added unit tests

 

## Documentation
N/A

### CODE REVIEW

File: backend/routes/Environment.types.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+export interface ExternallCapability extends CapabilityBase {
+    source: 'input' | 'secret';
+    blueprintDefault: boolean;
+}
+
+export interface PostEnvironmentBlueprintRequestBody {
+    description: string;
+    capabilities: ((CapabilityBase & { source: 'static' }) | ExternallCapability)[];
```
Comment (PatrycjaKubiczek) [inline backend/routes/Environment.types.ts:13]:
```suggestion
    capabilities: ((CapabilityBase & { source: 'static' }) | ExternalCapability)[];
```

File: backend/routes/Environment.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import express from 'express';
+import _ from 'lodash';
+import { renderBlueprint } from '../handler/EnvironmentHandler';
+import type {
+    ExternallCapability,
```
Comment (PatrycjaKubiczek) [inline backend/routes/Environment.ts:5]:
```suggestion
    ExternalCapability,
```

File: backend/routes/Environment.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+router.post<never, PostEnvironmentBlueprintResponse, PostEnvironmentBlueprintRequestBody>(
+    '/blueprint',
+    (req, res, next) => {
+        const environmentData = req.body;
+        const environmentRenderParams: EnvironmentRenderParams = {
+            description: environmentData.description,
+            inputs: _(environmentData.capabilities.filter(capability => capability.source !== 'static'))
+                .keyBy('name')
+                .mapValues(capability =>
+                    (<ExternallCapability>capability).blueprintDefault
```
Comment (PatrycjaKubiczek) [inline backend/routes/Environment.ts:23]:
```suggestion
                    (<ExternalCapability>capability).blueprintDefault
```

File: backend/routes/Environment.types.ts (hunk 1 above)
Comment (qooban) [inline backend/routes/Environment.types.ts:13]: Optional:

Maybe we could name `(CapabilityBase & { source: 'static' })` type?

File: backend/handler/EnvironmentHandler.types.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export interface EnvironmentRenderParams {
+    description: string;
+    inputs: Record<string, null | { defaultSource: 'static' | 'secret'; defaultValue: string }>;
+    labels: Record<string, string>;
+    capabilities: Record<string, { source: 'static' | 'input'; value: string }>;
```
Comment (qooban) [inline backend/handler/EnvironmentHandler.types.ts:5]: Optional:

Probably we could somehow reuse `ExternallCapability['source']` here.

**Review by qooban — APPROVED** [review]
Only optional comments, so approving.

File: backend/routes/Environment.types.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+interface CapabilityBase {
+    name: string;
+    value: string;
+}
+
+export interface ExternallCapability extends CapabilityBase {
```
[the remark this answers is not in the document]
Comment (kubama) [reply backend/routes/Environment.types.ts]: Done

File: backend/routes/Environment.types.ts (hunk 1 above)
Comment (kubama) [reply backend/routes/Environment.types.ts]: Done

File: backend/routes/Environment.ts (hunk 2 above)
Comment (kubama) [reply backend/routes/Environment.ts]: Done

File: backend/routes/Environment.ts (hunk 3 above)
Comment (kubama) [reply backend/routes/Environment.ts]: Done

File: backend/routes/Environment.types.ts (hunk 1 above)
Comment (kubama) [reply backend/routes/Environment.types.ts]: Done

File: backend/handler/EnvironmentHandler.types.ts (hunk 4 above)
Comment (kubama) [reply backend/handler/EnvironmentHandler.types.ts:5]: Actually the set of possible values is different

**Pull request merged**